saleor/graphql/core/types/taxes.py

```diff
@@ -383,7 +384,13 @@ def calculate_checkout_discounts(checkout_info):
                 checkout = checkout_info.checkout
                 discount_name = checkout.discount_name
                 return (
-                    [{"name": discount_name, "amount": checkout.discount}]
+                    [
+                        {
+                            "name": discount_name,
+                            "amount": checkout.discount,
+                            "type": checkout_info.voucher.type,
```

Comment:
Having discount, doesnt always mean that you have a voucher. See the screenshot with simple search over discount_name. So setting the `type` based on voucher will raise unhandled exception.
<img width="633" alt="image" src="https://github.com/saleor/saleor/assets/23282005/39114d87-70a6-4553-be3d-9a65a1771005">